corehq/messaging/smsbackends/connectid/views.py

```diff
@@ -27,3 +33,55 @@ def receive_message(request, *args, **kwargs):
     )
     process_incoming(msg, phone_obj)
     return HttpResponse("")
+
+
+@csrf_exempt
+@connectid_token_auth
+def connectid_messaging_key(request, domain):
+    link = get_object_or_404(ConnectIDUserLink, commcare_user=request.user, domain=request.domain)
+    key = generate_aes_key().decode("utf-8")
+    messaging_key, _ = ConnectIDMessagingKey.objects.get_or_create(
+        connectid_user_link=link, domain=request.domain, active=True, defaults={"key": key}
+    )
+    return JsonResponse({"key": messaging_key.key})
+
+
+@csrf_exempt
+@require_POST
+@validate_request_hmac("CONNECTID_SECRET_KEY")
+def update_connectid_messaging_consent(request, domain):
+    username = request.POST.get("username")
```

Comment:
I think this also wants `channel_id` instead of username

Reply:
Fixed. [3cb58a1](https://github.com/dimagi/commcare-hq/pull/35273/commits/3cb58a13326649332d7ed85619397b072f55ae2c)